astropy/utils/shapes.py

```diff
@@ -85,32 +87,32 @@ def reshape(self, *args, **kwargs):
         the shape attribute (note: this may not be implemented for all classes
         using ``NDArrayShapeMethods``).
         """
-        return self._apply("reshape", *args, **kwargs)
+        return self._apply("reshape", *newshape, order=order)
 
-    def ravel(self, *args, **kwargs):
+    def ravel(self, order: Literal["C", "F", "A", "K"] = "C") -> Self:
         """Return an instance with the array collapsed into one dimension.
 
         Parameters are as for :meth:`~numpy.ndarray.ravel`. Note that it is
         not always possible to unravel an array without copying the data.
         If you want an error to be raise if the data is copied, you should
         should assign shape ``(-1,)`` to the shape attribute.
         """
-        return self._apply("ravel", *args, **kwargs)
+        return self._apply("ravel", order=order)
 
-    def flatten(self, *args, **kwargs):
+    def flatten(self, order: Literal["C", "F", "A", "K"] = "C") -> Self:
         """Return a copy with the array collapsed into one dimension.
 
         Parameters are as for :meth:`~numpy.ndarray.flatten`.
         """
-        return self._apply("flatten", *args, **kwargs)
+        return self._apply("flatten", order=order)
 
-    def transpose(self, *args, **kwargs):
+    def transpose(self, *axes: int | tuple[int, ...] | None) -> Self:
```

Comment:
numpy itself does a `from typing import SupportsIndex`; see https://github.com/numpy/numpy/blob/2d2ff2501ad8a558c9ea402bf7b1a48e3f676fed/numpy/__init__.pyi#L1574-L1577

Reply:
`int` is a subtype of `SupportsIndex`, so in principle using `int` means our types are stricter than the `numpy` ones. However, the use of instances of `SupportsIndex` that are not `int` instances is very rare, so I don't consider this to make a difference in practice. Furthermore, I expect that there are many more `astropy` users who know what `int` is that there are users who know what `SupportsIndex` is.